### ISSUE TITLE
 Support for incremental learning of sklearn.neural_network.MLPRegressor

### ISSUE THREAD

**Issue by palaiyacw** [jpmml/jpmml-sklearn#31]
I am using the following code to train the `sklearn.neural_network.MLPRegressor` and finally converting it to PMML. Is there any way by means of which I can train my model with incremental data in the current version of the library?
        
```
              predictivescore_df = pandas.read_csv(filename)	

		clf = MLPRegressor(solver='lbfgs', alpha=1e-5,
		                    hidden_layer_sizes=(5), random_state=1, max_iter=100)

		pipeline = PMMLPipeline([
		  ('clf', clf)
		])
		pipeline.fit(predictivescore_df[predictivescore_df.columns.difference(["label"])], predictivescore_df["label"])			
		sklearn2pmml(pipeline, "PredictiveModel.pmml")
```

**Comment by vruusmann** [comment]
Python class `sklearn.neural_network.MLPRegressor` is fully supported.

The incremental learning happens inside Scikit-Learn framework, and does not affect the PMML layer in any way.

**Comment by palaiyacw** [comment]
@vruusmann  I am doing like this instead of calling `pipeline.fit`. I used `partial_fit` on `MLPRegressor` after unpickling the model saved previously.  But now the problem is alias of DataField comes out to be x1, x2 , x3 like this `<DataField name="x1" optype="continuous" dataType="double"/>`  instead of actual column names definedin csv. Could you please suggest how to achieve the same? 


```
predictivescore_df = pandas.read_csv(filename)	

clf = MLPRegressor(solver='lbfgs', alpha=1e-5,
                hidden_layer_sizes=(5), random_state=1, max_iter=100)

clf = pickle.load(open("PreviouslySavedPklFile", 'rb'))
clf.partial_fit(predictivescore_df[predictivescore_df.columns.difference(["Label"])], predictivescore_df["Label"])

pipeline = PMMLPipeline([
('clf', clf)
])
			
sklearn2pmml(pipeline, "PredictiveModel.pmml")
```

**Comment by zepen** [comment]
> @vruusmann I am doing like this instead of calling `pipeline.fit`. I used `partial_fit` on `MLPRegressor` after unpickling the model saved previously. But now the problem is alias of DataField comes out to be x1, x2 , x3 like this `<DataField name="x1" optype="continuous" dataType="double"/>` instead of actual column names defined in csv. Could you please suggest how to achieve the same? Reason being, I can't call `partial_fit` on pipeline
> 
> ```
> predictivescore_df = pandas.read_csv(filename)	
> 
> clf = MLPRegressor(solver='lbfgs', alpha=1e-5,
>                 hidden_layer_sizes=(5), random_state=1, max_iter=100)
> 
> clf = pickle.load(open("PreviouslySavedPklFile", 'rb'))
> clf.partial_fit(predictivescore_df[predictivescore_df.columns.difference(["Label"])], predictivescore_df["Label"])
> 
> pipeline = PMMLPipeline([
> ('clf', clf)
> ])
> 			
> sklearn2pmml(pipeline, "PredictiveModel.pmml")
> ```

hello，i have same problem. Do you find anyway to solve your question?

**Comment by vruusmann** [comment]
@zepen If the `PMMLPipeline.fit(X, y)` method was not called, then `PMMLPipeline.active_fields` (feature names) and `PMMLPipeline.target_fields` (label name) remain uninitialized. However, they can be set manually:

```python
pipeline = PMMLPipeline([
  ('clf', clf)
])
pipeline.active_fields = ["input1", "input2", "input3"]
pipeline.target_fields = ["prediction"]
```

**Comment by zepen** [comment]
> @zepen If the `PMMLPipeline.fit(X, y)` method was not called, then `PMMLPipeline.active_fields` (feature names) and `PMMLPipeline.target_fields` (label name) remain uninitialized. However, they can be set manually:
> 
> ```python
> pipeline = PMMLPipeline([
>   ('clf', clf)
> ])
> pipeline.active_fields = ["input1", "input2", "input3"]
> pipeline.target_fields = ["prediction"]
> ```

think a lot
